app-modules/knowledge-base/tests/KnowledgeBaseCategory/EditKnowledgeBaseCategoryTest.php

```diff
@@ -140,3 +145,28 @@
 
     assertEquals($request['name'], $knowledgeBaseCategory->fresh()->name);
 });
+
+test('can attach subcategories into categories', function () {
+    $user = User::factory()->licensed(LicenseType::cases())->create();
+
+    $user->givePermissionTo('knowledge_base_category.view-any');
+    $user->givePermissionTo('knowledge_base_category.*.update');
+
+    $knowledgeBaseCategory = KnowledgeBaseCategory::factory()->create();
+
+    $knowledgeBaseSubCategory = KnowledgeBaseCategory::factory()->create();
+
+    livewire(SubCategoriesRelationManager::class, [
+        'ownerRecord' => $knowledgeBaseCategory,
+        'pageClass' => EditKnowledgeBaseCategory::class,
+    ])
+        ->callTableAction(
+            AssociateAction::class,
+            data: ['recordId' => $knowledgeBaseSubCategory->getKey()]
+        )->assertSuccessful();
+
+    expect($knowledgeBaseCategory->refresh())
+        ->subCategories
+        ->pluck('id')
+        ->toContain($knowledgeBaseSubCategory->getKey());
```

Comment:
I would write an expectation before your livewire component call asserting that it does NOT contain the key first. To be sure that it was the code executed that put it into this state.